app/views/components/_email-signup-section.html.erb

```diff
@@ -0,0 +1,45 @@
+<%
+  details ||= {}
+  link_text ||= false
+  link_href ||= false
+  intro ||= false
+  full_width ||= false
+  track_action ||= false
+  track_label ||= false
+  transparent_background ||= false
+  css_classes ||= false
+  classes ||= []
+  tracking_data = track_action ? {
+    module: "track-click",
+    "track-category": "emailAlertLinkClicked",
+    "track-action": track_action,
+    "track-label": track_label
+  } : {}
+
+  classes << css_classes if css_classes
+  classes << "app-c-email-signup--full-width" if full_width
+  classes << "app-c-email-signup--no-intro" if !intro
+  classes << "app-c-email-signup--transparent_background" if transparent_background
+
+  #  if full_width is true, wrap content with govuk layout classes for full width styling
+  def full_width_wrapper(is_full_width, &block)
+    str = is_full_width ? "<div class='govuk-width-container'><div class='govuk-grid-row'><div class='govuk-grid-column-two-thirds'>" : ""
+    str += capture(&block)
+    str += is_full_width ?  "</div></div></div>" : ""
+    raw str
+  end
+%>
+<% if link_text && link_href %>
+  <div class="app-c-email-signup dont-print <%= classes.join(' ') %>">
+    <%= full_width_wrapper(full_width) do %>
+      <svg  class="app-c-email-signup__icon" aria-hidden="true" xmlns="http://www.w3.org/2000/svg" height="18" width="18" viewBox="0 0 459.334 459.334">
+        <path fill="black" d="M177.216 404.514c-.001.12-.009.239-.009.359 0 30.078 24.383 54.461 54.461 54.461s54.461-24.383 54.461-54.461c0-.12-.008-.239-.009-.359H175.216zM403.549 336.438l-49.015-72.002v-89.83c0-60.581-43.144-111.079-100.381-122.459V24.485C254.152 10.963 243.19 0 229.667 0s-24.485 10.963-24.485 24.485v27.663c-57.237 11.381-100.381 61.879-100.381 122.459v89.83l-49.015 72.002a24.76 24.76 0 0 0 20.468 38.693H383.08a24.761 24.761 0 0 0 20.469-38.694z"/>
+      </svg>
+      <%= content_tag(:h3, intro, class: "govuk-heading-s app-c-email-signup__title") if intro %>
```

Comment:
Is H3 always appropriate here, or is it worth adding an option to control the heading level?

Reply:
I think `h3` is appropriate for most instances of this component, however I've added the option to pass through `heading_level` since I think for the hub pages using `h2` would actually be more accurate (let me know what you think though)

Reply:
Sounds good. I should have pointed out sooner, the shared helper in the gem handles heading levels as well, might be worth taking a look.